src/main/java/entity/Goal.java

```diff
@@ -16,7 +16,8 @@ public class Goal {
     private final Task targetTask;
     private final TimePeriod timePeriod;
     private final int frequency;
-    private final List<LocalDate> completionDates;
+    private List<LocalDate> completionDates;
```

Comment:
```suggestion
    private final List<LocalDate> completionDates;
```
Keep this field final for immutability. The list contents can still be modified through the recordCompletion() method, but the reference itself should not change.